pkg/secretless/entrypoint/entrypoint.go

```diff
@@ -90,6 +94,8 @@ func handlePerformanceProfiling(profileType string) {
 
 	// Start profiling
 	perfProfile := profile.New(profileType)
-	signal.StopOnExitSignal(perfProfile)
+
+	// Wrap the exit signal handler in go routine so it won't block
+	go func() { signal.StopOnExitSignal(perfProfile) }()
```

Comment:
nit: I know the codeblock is tiny but this might look better split into multiple lines